uki-generate

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3 -I
+import os
+import rpm
+import subprocess
+import sys
+from typing import List, Tuple
+
+def main(args):
+    alignment_mask = (1 << 21) - 1
+    if len(args) != 7 or args[1] != '--':
+        print(f"Usage: uki-generate -- HYPERVISOR CONFIG KERNEL INITRAMFS OUTPUT", file=sys.stderr)
+        sys.exit(1)
+    _, _, hyp, cfg, kern, initramfs, out = args
+    if hyp[0] != '/':
+        hyp = './' + hyp
+    if out[0] != '/':
+        out = './' + out
+    def round_to_next(f: int) -> int:
+        if f > (0xffffffffffffffff & ~alignment_mask):
+            raise ValueError("Address overflow")
+        return (f + alignment_mask) & ~alignment_mask
+    base_address = 0xffff82d042000000
+    kernel_vma = round_to_next(base_address + os.stat(cfg).st_size)
+    initramfs_vma = round_to_next(kernel_vma + os.stat(kern).st_size)
+    cmdline = [
+        "objcopy",
+        f"--section-alignment={alignment_mask + 1}",
+        "--remove-section=.gnu.*",
+        "--remove-section=.buildid",
+        "--strip-debug",
+        f"--add-section=.config={cfg}",
+        f"--change-section-vma=.config={base_address}",
+        f"--add-section=.kernel={kern}",
+        f"--change-section-vma=.kernel={kernel_vma}",
+        f"--add-section=.ramdisk={initramfs}",
+        f"--change-section-vma=.ramdisk={initramfs_vma}",
+        "--",
+        hyp,
+        out,
+    ]
+    subprocess.check_call(cmdline, stdin=subprocess.DEVNULL, stdout=subprocess.DEVNULL)
+if __name__ == '__main__':
+    try:
+        main(sys.argv)
+    except subprocess.CalledProcessError as e:
```

Comment:
```suggestion
    except (subprocess.CalledProcessError, ValueError) as e:
        print(str(e))
```

Reply:
That should not happen.

Reply:
That should not happen.

Reply:
You raise ValueError in the code so it can happen and we need the error message.

Reply:
`ValueError` is thrown if one of the file sizes is nearly 2<sup>64</sup> bytes.  This is absolutely absurd and I don’t think we need to handle this situation.

Reply:
If something really shouldn't happen, make it an assert. Other exceptions should be properly handled.